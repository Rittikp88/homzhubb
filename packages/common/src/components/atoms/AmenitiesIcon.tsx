import React from 'react';
import { StyleProp, StyleSheet, View, ViewStyle } from 'react-native';
import Icon from '@homzhub/common/src/assets/icon';
import { theme } from '@homzhub/common/src/styles/theme';
import { Text } from '@homzhub/common/src/components/atoms/Text';
import { Divider } from '@homzhub/common/src/components/atoms/Divider';

interface IProps {
  direction: 'row' | 'column';
  icon: string;
  iconSize?: number;
  iconColor?: string;
  label: string;
  isLastIndex?: boolean;
  containerStyle?: StyleProp<ViewStyle>;
}

const AmenitiesIcon = (props: IProps): React.ReactElement => {
  const {
    direction,
    icon,
    iconSize = 40,
    iconColor = theme.colors.darkTint3,
    label,
    isLastIndex = false,
    containerStyle,
  } = props;

  const renderText = (): React.ReactElement => {
    const labelStyle = direction === 'column' ? styles.columnLabel : styles.rowLabel;
    return (
      <Text type="small" textType="regular" style={[styles.label, labelStyle]}>
        {label}
      </Text>
    );
  };

  const renderColumn = (): React.ReactElement => {
    return (
      <>
        <View style={styles.columnContainer}>
          <Icon name={icon} size={iconSize} color={iconColor} />
          {renderText()}
        </View>
        {!isLastIndex && direction === 'column' && <Divider containerStyles={styles.divider} />}
      </>
    );
  };

  const renderRow = (): React.ReactElement => {
    return (
      <View style={[styles.rowContainer, containerStyle]}>
        <Icon name={icon} size={iconSize} color={iconColor} />
        {renderText()}
      </View>
    );
  };

  return direction === 'column' ? renderColumn() : renderRow();
};

export { AmenitiesIcon };

const styles = StyleSheet.create({
  columnContainer: {
    flexDirection: 'column',
    justifyContent: 'space-around',
    alignItems: 'center',
    marginHorizontal: 10,
  },
  rowContainer: {
    flexDirection: 'row',
    alignItems: 'center',
  },
  label: {
    color: theme.colors.darkTint3,
  },
  columnLabel: {
    alignItems: 'center',
  },
  rowLabel: {
    marginStart: 6,
  },
  divider: {
    borderWidth: 1,
    borderColor: theme.colors.background,
    height: 25,
  },
});
