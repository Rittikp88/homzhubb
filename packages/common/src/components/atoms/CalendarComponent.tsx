import React, { Component } from 'react';
import { Calendar, DateObject } from 'react-native-calendars';
import { View, FlatList, TouchableOpacity, StyleSheet, StyleProp, TextStyle, ViewStyle } from 'react-native';
import moment from 'moment';
import { DateFormats, DateUtils, MonthNames } from '@homzhub/common/src/utils/DateUtils';
import { PlatformUtils } from '@homzhub/common/src/utils/PlatformUtils';
import Icon, { icons } from '@homzhub/common/src/assets/icon';
import { theme } from '@homzhub/common/src/styles/theme';
import { Divider, Button, Text, Label, WithShadowView } from '@homzhub/common/src/components';

interface ICalendarProps {
  onSelect: (day: string) => void;
  selectedDate: string;
}

interface ICalendarState {
  isMonthView: boolean;
  selectedDate: string;
  day: string;
  month: number;
  year: string;
}

export class CalendarComponent extends Component<ICalendarProps, ICalendarState> {
  public constructor(props: ICalendarProps) {
    super(props);
    const { selectedDate } = props;
    this.state = {
      isMonthView: false,
      selectedDate,
      day: moment().format(DateFormats.DD),
      month: new Date().getMonth(),
      year: moment().format(DateFormats.YYYY),
    };
  }

  public render(): React.ReactNode {
    const { isMonthView } = this.state;
    return (
      <View>
        {this.renderHeader()}
        {isMonthView ? this.renderMonthView() : this.renderCalendar()}
      </View>
    );
  }

  private renderHeader = (): React.ReactElement => {
    const { year, month, isMonthView, selectedDate } = this.state;
    const newMonth = moment(selectedDate).month();
    const newYear = moment(selectedDate).year();
    const updateMonth = DateUtils.getFullMonthName(newMonth || month, DateFormats.MMMM);

    return (
      <>
        <View style={styles.headerContainer}>
          <Icon
            name={icons.leftArrow}
            onPress={this.handleBackPress}
            size={22}
            color={month === moment().month() ? theme.colors.disabled : theme.colors.primaryColor}
          />
          <Text
            type="small"
            textType="semiBold"
            onPress={this.handleMonthPress}
            style={StyleSheet.flatten([customStyles.headerTitle(isMonthView)])}
          >
            {`${updateMonth} ${newYear || year}`}
          </Text>
          <Icon name={icons.rightArrow} size={22} color={theme.colors.primaryColor} onPress={this.handleNextPress} />
        </View>
        <Divider />
      </>
    );
  };

  private renderMonthView = (): React.ReactNode => {
    return (
      <FlatList
        data={MonthNames}
        renderItem={this.renderItem}
        contentContainerStyle={styles.listContent}
        keyExtractor={this.renderKeyExtractor}
        numColumns={4}
      />
    );
  };

  private renderKeyExtractor = (item: string, index: number): string => {
    return `${item}-${index}`;
  };

  private renderItem = ({ item, index }: { item: string; index: number }): React.ReactElement => {
    const onPressItem = (): void => this.onSelectMonth(item, index);
    const { month } = this.state;
    const isSelected = month === index;
    return (
      <TouchableOpacity
        key={index}
        style={StyleSheet.flatten([customStyles.renderItemView(isSelected)])}
        onPress={onPressItem}
      >
        <Label type="large" style={StyleSheet.flatten([customStyles.renderItemTitle(isSelected)])}>
          {item}
        </Label>
      </TouchableOpacity>
    );
  };

  private renderCalendar = (): React.ReactElement => {
    const { day, month, year, selectedDate } = this.state;
    const updateMonth = DateUtils.getFullMonthName(month, DateFormats.MMM);
    const date =
      selectedDate || DateUtils.getFormattedDate(day, updateMonth, year, DateFormats.YYYYMMMDD).toDateString();
    return (
      <>
        <Calendar
          hideArrows
          // @ts-ignore
          renderHeader={this.headerView}
          minDate={new Date()}
          current={date}
          key={date}
          style={styles.calendarStyle}
          onDayPress={this.onDayPress}
          markingType="custom"
          theme={{}}
          markedDates={{
            [date]: {
              customStyles: {
                container: {
                  backgroundColor: theme.colors.primaryColor,
                  borderRadius: 4,
                },
                text: {
                  color: theme.colors.white,
                },
              },
            },
          }}
        />
        <WithShadowView outerViewStyle={styles.shadowView}>
          <Button type="primary" title="Select" containerStyle={styles.buttonStyle} onPress={this.handleSelect} />
        </WithShadowView>
      </>
    );
  };

  private onSelectMonth = (item: string, index: number): void => {
    const { year } = this.state;
    this.setState({ month: index, isMonthView: false });
    this.getSelectedDate(index, Number(year));
  };

  private onDayPress = (day: DateObject): void => {
    this.setState({ selectedDate: day.dateString });
  };

  private handleSelect = (): void => {
    const { onSelect } = this.props;
    const { selectedDate } = this.state;
    onSelect(selectedDate);
  };

  private headerView = (): null => {
    return null;
  };

  private handleBackPress = (): void => {
    const { year, isMonthView, month } = this.state;

    if (month === moment().month()) {
      return;
    }

    if (isMonthView) {
      const previousYear = Number(year) - 1;
      this.getSelectedDate(month, previousYear);
      this.setState({ year: previousYear.toString() });
    } else {
      const previousMonth = month - 1;
      this.getSelectedDate(previousMonth, Number(year));
      this.setState({ month: previousMonth });
    }
  };

  private handleNextPress = (): void => {
    const { year, isMonthView, month } = this.state;

    if (isMonthView) {
      const nextYear = Number(year) + 1;
      this.getSelectedDate(month, nextYear);
      this.setState({ year: nextYear.toString() });
    } else {
      const nextMonth = month + 1;
      this.getSelectedDate(nextMonth, Number(year));
      this.setState({ month: nextMonth });
    }
  };

  private getSelectedDate = (month: number, year: number): void => {
    const { selectedDate } = this.props;
    const isNotSelected = moment(selectedDate).month() !== month || moment(selectedDate).year() !== year;
    if (isNotSelected) {
      this.setState({ selectedDate: '' });
    } else {
      this.setState({ selectedDate });
    }
  };

  private handleMonthPress = (): void => {
    const { isMonthView } = this.state;
    this.setState({ isMonthView: !isMonthView });
  };
}

const styles = StyleSheet.create({
  shadowView: {
    paddingTop: 10,
    marginBottom: PlatformUtils.isIOS() ? 20 : 0,
    paddingBottom: 0,
  },
  buttonStyle: {
    flex: 0,
    margin: 16,
  },
  headerContainer: {
    justifyContent: 'space-between',
    flexDirection: 'row',
    paddingVertical: 20,
    paddingHorizontal: 24,
  },
  listContent: {
    marginVertical: 16,
  },
  calendarStyle: {
    height: 310,
  },
});

const customStyles = {
  headerTitle: (isMonthView: boolean): StyleProp<TextStyle> => ({
    color: isMonthView ? theme.colors.darkTint2 : theme.colors.primaryColor,
  }),
  renderItemView: (isSelected: boolean): StyleProp<ViewStyle> => ({
    width: 80,
    marginVertical: 12,
    marginHorizontal: 10,
    alignItems: 'center',
    backgroundColor: isSelected ? theme.colors.primaryColor : theme.colors.white,
    borderRadius: 4,
  }),
  renderItemTitle: (isSelected: boolean): StyleProp<TextStyle> => ({
    paddingVertical: 6,
    color: isSelected ? theme.colors.white : theme.colors.darkTint2,
  }),
};
