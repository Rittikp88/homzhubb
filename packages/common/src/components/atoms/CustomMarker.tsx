import React from 'react';
import { View, StyleSheet } from 'react-native';
import { theme } from '@homzhub/common/src/styles/theme';

interface IProps {
  selected: boolean;
}

const CustomMarker = (props: IProps): React.ReactElement => {
  const { selected } = props;

  if (!selected) {
    return <View style={styles.marker} />;
  }

  return (
    <View style={styles.selectedMarker}>
      <View style={styles.marker} />
    </View>
  );
};

const styles = StyleSheet.create({
  marker: {
    width: 16,
    height: 16,
    borderRadius: 16 / 2,
    backgroundColor: theme.colors.primaryColor,
  },
  selectedMarker: {
    justifyContent: 'center',
    alignItems: 'center',
    width: 40,
    height: 40,
    borderRadius: 40 / 2,
    backgroundColor: theme.colors.markerOpacity,
  },
});

const memoizedComponent = React.memo(CustomMarker);
export { memoizedComponent as CustomMarker };
