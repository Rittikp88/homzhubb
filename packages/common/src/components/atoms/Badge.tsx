import React from 'react';
import { StyleProp, StyleSheet, View, ViewStyle } from 'react-native';
import { theme } from '@homzhub/common/src/styles/theme';
import { FontWeightType, Label } from '@homzhub/common/src/components/atoms/Text';

interface IProps {
  title: string;
  badgeColor: string;
  badgeStyle?: StyleProp<ViewStyle>;
  textType?: FontWeightType;
}

export const Badge = (props: IProps): React.ReactElement => {
  const { badgeColor, badgeStyle, title, textType } = props;
  return (
    <View style={[styles.badge, badgeStyle, { backgroundColor: badgeColor }]}>
      <Label type="regular" textType={textType} style={styles.title}>
        {title}
      </Label>
    </View>
  );
};

const styles = StyleSheet.create({
  badge: {
    borderRadius: 2,
    paddingHorizontal: 6,
  },
  title: {
    color: theme.colors.white,
    alignSelf: 'center',
  },
});
