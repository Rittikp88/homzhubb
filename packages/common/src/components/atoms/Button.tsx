import React from 'react';
import {
  GestureResponderEvent,
  ImageStyle,
  StyleProp,
  StyleSheet,
  TextStyle,
  TouchableOpacity,
  ViewStyle,
} from 'react-native';
import { theme } from '@homzhub/common/src/styles/theme';
import Icon from '@homzhub/common/src/assets/icon';
import { FontWeightType, Label, Text, TextFieldType, TextSizeType } from '@homzhub/common/src/components/atoms/Text';

export type ButtonType = 'primary' | 'secondary' | 'secondaryOutline';

export interface IButtonProps {
  type: ButtonType;
  node?: React.ReactNode;
  children?: React.ReactNode;
  title?: string;
  onPress?: (event?: GestureResponderEvent) => void;
  disabled?: boolean;
  activeOpacity?: number;
  titleStyle?: StyleProp<TextStyle>;
  containerStyle?: StyleProp<ViewStyle>;
  textType?: TextFieldType;
  textSize?: TextSizeType;
  fontType?: FontWeightType;
  icon?: string;
  iconSize?: number;
  iconColor?: string;
  iconStyle?: StyleProp<ImageStyle>;
  testID?: string;
}

export class Button extends React.PureComponent<IButtonProps> {
  public render = (): React.ReactElement => {
    const {
      onPress,
      disabled = false,
      activeOpacity = 0.5,
      title,
      icon,
      iconSize,
      iconColor,
      iconStyle,
      testID,
      node,
      children,
    } = this.props;
    return (
      <TouchableOpacity
        activeOpacity={activeOpacity}
        onPress={onPress}
        disabled={disabled}
        style={this.getContainerStyle()}
        testID={testID}
      >
        {title && this.getTextField()}
        {!!icon && <Icon name={icon} size={iconSize} color={iconColor} style={iconStyle} />}
        {node && node}
        {children && children}
      </TouchableOpacity>
    );
  };

  private getTextField = (): React.ReactElement => {
    const { textType, title, textSize = 'small', fontType = 'semiBold' } = this.props;
    let TextField = Text;

    if (textType === 'label') {
      TextField = Label;
    }

    return (
      <TextField type={textSize} textType={fontType} style={this.getTextStyle()}>
        {title}
      </TextField>
    );
  };

  private getContainerStyle = (): ViewStyle => {
    const { disabled, type, containerStyle = {} } = this.props;
    let themedStyle: StyleProp<ViewStyle> = styles.primary;
    if (disabled) {
      themedStyle = styles.disabled;
    }

    if (type === 'secondary' && !disabled) {
      themedStyle = styles.secondary;
    }

    if (type === 'secondaryOutline' && !disabled) {
      themedStyle = styles.secondaryOutline;
    }
    return StyleSheet.flatten([styles.container, themedStyle, containerStyle]);
  };

  private getTextStyle = (): TextStyle => {
    const { type, disabled = false, titleStyle = {} } = this.props;
    let themedStyle = {};
    if (type === 'secondary' && !disabled) {
      themedStyle = { color: theme.colors.active };
    }
    return StyleSheet.flatten([styles.textStyle, themedStyle, titleStyle]);
  };
}

const styles = StyleSheet.create({
  container: {
    ...theme.globalStyles.center,
    borderRadius: 4,
  },
  textStyle: {
    textAlign: 'center',
    color: theme.colors.white,
    marginVertical: 12,
    marginHorizontal: 36,
  },
  disabled: {
    backgroundColor: theme.colors.disabled,
  },
  primary: {
    backgroundColor: theme.colors.primaryColor,
  },
  secondary: {
    borderWidth: 1,
    borderColor: theme.colors.active,
    backgroundColor: theme.colors.white,
  },
  secondaryOutline: {
    borderWidth: 0.5,
    borderColor: theme.colors.white,
  },
});
