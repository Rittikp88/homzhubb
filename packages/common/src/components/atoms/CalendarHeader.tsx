import React from 'react';
import { StyleSheet, View } from 'react-native';
import moment from 'moment';
import Icon, { icons } from '@homzhub/common/src/assets/icon';
import { theme } from '@homzhub/common/src/styles/theme';
import { Divider } from '@homzhub/common/src/components/atoms/Divider';
import { Text } from '@homzhub/common/src/components/atoms/Text';

interface IProps {
  isAllowPastDate?: boolean;
  headerTitle?: string;
  headerYear?: string;
  yearTitle?: string;
  isMonthView?: boolean;
  isYearView?: boolean;
  isCurrentMonth?: boolean;
  maxDate?: string;
  month?: number;
  onBackPress?: () => void;
  onNextPress?: () => void;
  onMonthPress?: () => void;
  onYearPress?: () => void;
}

const CalendarHeader = (props: IProps): React.ReactElement => {
  const {
    isAllowPastDate,
    maxDate,
    onBackPress,
    onNextPress,
    onMonthPress,
    onYearPress,
    isCurrentMonth,
    headerTitle,
    headerYear,
    isMonthView = false,
    isYearView = false,
    yearTitle,
    month,
  } = props;
  const isNextDisable =
    maxDate && (moment(maxDate).month() === moment().month() || (month && moment(maxDate).month() === month));
  return (
    <>
      <View style={styles.headerContainer}>
        <Icon
          name={icons.leftArrow}
          onPress={onBackPress}
          size={22}
          color={!isAllowPastDate && isCurrentMonth && !isYearView ? theme.colors.disabled : theme.colors.primaryColor}
        />
        {!isYearView && !isMonthView && (
          <Text type="small" textType="semiBold" onPress={onMonthPress} style={styles.headerTitle}>
            {headerTitle}
          </Text>
        )}
        {isMonthView && (
          <Text type="small" textType="semiBold" onPress={onYearPress} style={styles.headerTitle}>
            {headerYear}
          </Text>
        )}
        {isYearView && (
          <Text type="small" textType="semiBold" style={styles.headerTitle}>
            {yearTitle}
          </Text>
        )}
        <Icon
          name={icons.rightArrow}
          size={22}
          color={isNextDisable ? theme.colors.disabled : theme.colors.primaryColor}
          onPress={onNextPress}
        />
      </View>
      <Divider />
    </>
  );
};

export default CalendarHeader;

const styles = StyleSheet.create({
  headerContainer: {
    justifyContent: 'space-between',
    flexDirection: 'row',
    paddingVertical: 20,
    paddingHorizontal: 24,
  },
  headerTitle: {
    color: theme.colors.primaryColor,
  },
});
