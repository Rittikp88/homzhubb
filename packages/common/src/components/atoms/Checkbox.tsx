import React from 'react';
import { StyleProp, StyleSheet, TextStyle, TouchableOpacity, View, ViewStyle } from 'react-native';
import { theme } from '@homzhub/common/src/styles/theme';
import Icon, { icons } from '@homzhub/common/src/assets/icon';
import { Label, TextSizeType } from '@homzhub/common/src/components/atoms/Text';

interface ICheckboxOptions {
  selected: boolean;
  label: string;
  onToggle: () => void;
  containerStyle?: StyleProp<ViewStyle>;
  labelStyle?: StyleProp<TextStyle>;
  iconSize?: number;
  iconStyle?: object;
  iconSelectedStyle?: object;
  labelType?: TextSizeType;
}

const RNCheckbox = (props: ICheckboxOptions): React.ReactElement => {
  const {
    label = '',
    selected = false,
    containerStyle = {},
    labelStyle = {},
    iconSize = 22,
    iconStyle = {},
    iconSelectedStyle = {},
    labelType = 'large',
    onToggle,
  } = props;
  return (
    <View style={[styles.container, containerStyle]}>
      <TouchableOpacity onPress={onToggle}>
        <Icon
          name={selected ? icons.checkboxOn : icons.checkboxOff}
          size={iconSize}
          color={selected ? theme.colors.primaryColor : theme.colors.disabled}
          style={selected ? iconSelectedStyle : iconStyle}
        />
      </TouchableOpacity>
      <Label type={labelType} textType="regular" style={[styles.label, labelStyle]}>
        {label}
      </Label>
    </View>
  );
};

export { RNCheckbox };

const styles = StyleSheet.create({
  container: {
    flexDirection: 'row',
    alignItems: 'center',
  },
  label: {
    marginStart: 12,
  },
});
