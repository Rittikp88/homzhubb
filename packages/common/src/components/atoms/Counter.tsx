import React, { ReactElement, useEffect, useState } from 'react';
import { StyleProp, StyleSheet, View, ViewStyle } from 'react-native';
import { theme } from '@homzhub/common/src/styles/theme';
import Icon, { icons } from '@homzhub/common/src/assets/icon';
import { SVGUri } from '@homzhub/common/src/components/atoms/Svg';
import { Label, Text } from '@homzhub/common/src/components/atoms/Text';

interface ITitle {
  title: string;
  id?: number;
}

export interface ICounterProps {
  defaultValue: number;
  name?: ITitle;
  svgImage?: string;
  onValueChange: (count: number, id?: number) => void;
  maxCount?: number;
  minCount?: number;
  containerStyles?: StyleProp<ViewStyle>;
  testID?: string;
}

export const Counter = (props: ICounterProps): React.ReactElement => {
  const { onValueChange, defaultValue, name, svgImage, maxCount = 10, minCount = 0, containerStyles } = props;
  const [count, setCount] = useState(defaultValue);

  useEffect(() => {
    onValueChange(count, name?.id);
  }, [count]);

  const incrementCount = (): void => {
    if (count < maxCount) {
      setCount((prev) => prev + 1);
    }
  };

  const decrementCount = (): void => {
    if (count > minCount) {
      setCount((prev) => prev - 1);
    }
  };

  const renderCounter = (): ReactElement => {
    return (
      <View style={[styles.counterContainer, styles.rowStyle]}>
        <Icon name={icons.minus} color={theme.colors.primaryColor} size={20} onPress={decrementCount} />
        <Label type="large">{count}</Label>
        <Icon name={icons.plus} color={theme.colors.primaryColor} size={20} onPress={incrementCount} />
      </View>
    );
  };

  return (
    <View style={[styles.rowStyle, containerStyles]}>
      <View style={styles.imageContainer}>
        {svgImage && <SVGUri height={20} width={20} uri={svgImage} />}
        {name && (
          <Text style={styles.textStyle} type="small">
            {name.title}
          </Text>
        )}
      </View>
      {renderCounter()}
    </View>
  );
};

const styles = StyleSheet.create({
  rowStyle: {
    flexDirection: 'row',
    justifyContent: 'space-between',
    alignItems: 'center',
  },
  counterContainer: {
    borderColor: theme.colors.darkTint6,
    borderWidth: 1,
    borderRadius: 4,
    width: 100,
    padding: 4,
  },
  imageContainer: {
    flexDirection: 'row',
    alignItems: 'center',
  },
  textStyle: {
    marginLeft: 12,
  },
});
