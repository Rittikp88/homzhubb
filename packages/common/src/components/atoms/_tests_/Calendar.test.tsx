// @ts-nocheck
import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import { CalendarComponent } from '@homzhub/common/src/components/atoms/CalendarComponent';

jest.mock('@homzhub/common/src/services/storage/StorageService', () => 'StorageService');
jest.mock('@react-native-community/google-signin', () => {});

const createTestProps = (testProps: any): object => ({
  onSelect: jest.fn(),
  selectedDate: '2020-06-22',
  ...testProps,
});
let props: any;

describe('CalendarComponent', () => {
  it('should match snapshot when isMonthView is true', () => {
    props = createTestProps({});
    const wrapper = shallow(<CalendarComponent {...props} />);
    const instance = wrapper.instance();
    instance.handleMonthPress();
    instance.handleBackPress();
    instance.handleNextPress();
    instance.onSelectMonth(6, 0);
    instance.onDayPress({ dateString: '2020-06-19' });
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('should match snapshot when isMonthView is false', () => {
    props = createTestProps({
      selectedDate: '',
    });
    const wrapper = shallow(<CalendarComponent {...props} />);
    const instance = wrapper.instance();
    instance.handleSelect();
    instance.handleBackPress();
    instance.handleNextPress();
    instance.setState({ month: 6 });
    expect(instance).toMatchSnapshot();
  });
});
