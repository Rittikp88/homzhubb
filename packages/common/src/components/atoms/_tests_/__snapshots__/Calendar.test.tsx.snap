// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CalendarComponent should match snapshot when isMonthView is false 1`] = `
CalendarComponent {
  "context": Object {},
  "getSelectedDate": [Function],
  "handleBackPress": [Function],
  "handleMonthPress": [Function],
  "handleNextPress": [Function],
  "handleSelect": [Function],
  "headerView": [Function],
  "onDayPress": [Function],
  "onSelectMonth": [Function],
  "props": Object {
    "onSelect": [MockFunction] {
      "calls": Array [
        Array [
          "",
        ],
      ],
      "results": Array [
        Object {
          "type": "return",
          "value": undefined,
        },
      ],
    },
    "selectedDate": "",
  },
  "refs": Object {},
  "renderCalendar": [Function],
  "renderHeader": [Function],
  "renderItem": [Function],
  "renderKeyExtractor": [Function],
  "renderMonthView": [Function],
  "setState": [Function],
  "state": Object {
    "day": "07",
    "isMonthView": false,
    "month": 6,
    "selectedDate": "",
    "year": "2020",
  },
  "updater": Updater {
    "_callbacks": Array [],
    "_renderer": ReactShallowRenderer {
      "_context": Object {},
      "_didScheduleRenderPhaseUpdate": false,
      "_dispatcher": Object {
        "readContext": [Function],
        "useCallback": [Function],
        "useContext": [Function],
        "useDebugValue": [Function],
        "useDeferredValue": [Function],
        "useEffect": [Function],
        "useImperativeHandle": [Function],
        "useLayoutEffect": [Function],
        "useMemo": [Function],
        "useReducer": [Function],
        "useRef": [Function],
        "useResponder": [Function],
        "useState": [Function],
        "useTransition": [Function],
      },
      "_element": <CalendarComponent
        onSelect={
          [MockFunction] {
            "calls": Array [
              Array [
                "",
              ],
            ],
            "results": Array [
              Object {
                "type": "return",
                "value": undefined,
              },
            ],
          }
        }
        selectedDate=""
      />,
      "_firstWorkInProgressHook": null,
      "_forcedUpdate": false,
      "_instance": [Circular],
      "_isReRender": false,
      "_newState": null,
      "_numberOfReRenders": 0,
      "_renderPhaseUpdates": null,
      "_rendered": <View>
        <React.Fragment>
          <View
            style={
              Object {
                "flexDirection": "row",
                "justifyContent": "space-between",
                "paddingHorizontal": 24,
                "paddingVertical": 20,
              }
            }
          >
            <Icon
              allowFontScaling={false}
              color="#C9D6DF"
              name="left-arrow"
              onPress={[Function]}
              size={22}
            />
            <Text
              onPress={[Function]}
              style={
                Object {
                  "color": "#0084F8",
                }
              }
              textType="semiBold"
              type="small"
            >
              July 2020
            </Text>
            <Icon
              allowFontScaling={false}
              color="#0084F8"
              name="right-arrow"
              onPress={[Function]}
              size={22}
            />
          </View>
          <Divider />
        </React.Fragment>
        <React.Fragment>
          <Calendar
            current="Tue Jul 07 2020"
            enableSwipeMonths={false}
            hideArrows={true}
            markedDates={
              Object {
                "Tue Jul 07 2020": Object {
                  "customStyles": Object {
                    "container": Object {
                      "backgroundColor": "#0084F8",
                      "borderRadius": 4,
                    },
                    "text": Object {
                      "color": "#FFFFFF",
                    },
                  },
                },
              }
            }
            markingType="custom"
            minDate={2020-07-07T10:05:30.326Z}
            onDayPress={[Function]}
            renderHeader={[Function]}
            style={
              Object {
                "height": 310,
              }
            }
            theme={Object {}}
          />
          <WithShadowView
            outerViewStyle={
              Object {
                "marginBottom": 20,
                "paddingBottom": 0,
                "paddingTop": 10,
              }
            }
          >
            <Button
              containerStyle={
                Object {
                  "flex": 0,
                  "margin": 16,
                }
              }
              onPress={[Function]}
              title="Select"
              type="primary"
            />
          </WithShadowView>
        </React.Fragment>
      </View>,
      "_rendering": false,
      "_updater": [Circular],
      "_workInProgressHook": null,
    },
  },
}
`;

exports[`CalendarComponent should match snapshot when isMonthView is true 1`] = `
<View>
  <View
    style={
      Object {
        "flexDirection": "row",
        "justifyContent": "space-between",
        "paddingHorizontal": 24,
        "paddingVertical": 20,
      }
    }
  >
    <Icon
      allowFontScaling={false}
      color="#0084F8"
      name="left-arrow"
      onPress={[Function]}
      size={22}
    />
    <Text
      onPress={[Function]}
      style={
        Object {
          "color": "#0084F8",
        }
      }
      textType="semiBold"
      type="small"
    >
      June 2020
    </Text>
    <Icon
      allowFontScaling={false}
      color="#0084F8"
      name="right-arrow"
      onPress={[Function]}
      size={22}
    />
  </View>
  <Divider />
  <Calendar
    current="2020-06-19"
    enableSwipeMonths={false}
    hideArrows={true}
    key="2020-06-19"
    markedDates={
      Object {
        "2020-06-19": Object {
          "customStyles": Object {
            "container": Object {
              "backgroundColor": "#0084F8",
              "borderRadius": 4,
            },
            "text": Object {
              "color": "#FFFFFF",
            },
          },
        },
      }
    }
    markingType="custom"
    minDate={2020-07-07T10:05:30.294Z}
    onDayPress={[Function]}
    renderHeader={[Function]}
    style={
      Object {
        "height": 310,
      }
    }
    theme={Object {}}
  />
  <WithShadowView
    outerViewStyle={
      Object {
        "marginBottom": 20,
        "paddingBottom": 0,
        "paddingTop": 10,
      }
    }
  >
    <Button
      containerStyle={
        Object {
          "flex": 0,
          "margin": 16,
        }
      }
      onPress={[Function]}
      title="Select"
      type="primary"
    />
  </WithShadowView>
</View>
`;
